geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala

```diff
@@ -85,7 +85,11 @@ abstract class GeoMesaDataStore[DS <: GeoMesaDataStore[DS]](val config: GeoMesaD
     * (index tables and catalog table)
     * NB: We are *not* currently deleting the query table and/or query information.
     */
-  def delete(): Unit = adapter.deleteTables(getTypeNames.flatMap(getAllTableNames).distinct)
+  def delete(): Unit = {
+    val types = getTypeNames
+    val tables = if (types.length == 0) Array(config.catalog) else types
```

Comment:
seems like this isn't being used, did you mean to pass it in to adapter.deleteTables? it's a good catch, since previously if there was an empty catalog table it wouldn't delete it (although generally the catalog table isn't empty unless you've gone through and deleted all the feature types)

Reply:
sorry，my mistake,`tables` should be passed to adapter.deleteTables。

Reply:
> seems like this isn't being used, did you mean to pass it in to adapter.deleteTables? it's a good catch, since previously if there was an empty catalog table it wouldn't delete it (although generally the catalog table isn't empty unless you've gone through and deleted all the feature types)

sorry，my mistake,tables should be passed to adapter.deleteTables。